src/FNA3D_Driver_Vulkan.c

```diff
@@ -6765,6 +6764,16 @@ void VULKAN_QueryBegin(FNA3D_Renderer *driverData, FNA3D_Query *query)
 	FNAVulkanRenderer *renderer = (FNAVulkanRenderer*) driverData;
 	VulkanQuery *vulkanQuery = (VulkanQuery*) query;
 
+	/* Need to do this between passes */
+	EndPass(renderer);
+
+	renderer->vkCmdResetQueryPool(
```

Comment:
use vkResetCommandBuffer here instead, resetting the pool will break multiple frames in flight